util/utils.go

```diff
@@ -71,22 +71,54 @@ func UblockNetworkReader(
 		panic(err)
 	}
 
+	// TODO goroutines for only the Deserialize part might be nice.
+	// Need to sort the blocks though if you're doing that
 	for ; ; curHeight++ {
+		if err != nil {
+			panic(err)
+		}
+		block, err := ReadBlock(con)
+		if err != nil {
+			fmt.Println(err)
+			panic(err)
+		}
+
 		var ub UBlock
-		err = ub.Deserialize(con)
+		bReader := bytes.NewReader(block)
+		err = ub.Deserialize(bReader)
 		if err != nil {
 			if err == io.EOF {
 				close(blockChan)
 				break
 			}
```

Comment:
This should be moved to the error handling of ReadBlock, then if we get an EOF like in the github action right now it breaks and is done.

Reply:
Oops. Fix applied :)